test/e2e/serial/tests/workload_unschedulable.go

```diff
@@ -265,6 +265,38 @@ var _ = Describe("[serial][disruptive][scheduler] numaresources workload unsched
 				Expect(isFailed).To(BeTrue(), "pod %s/%s with scheduler %s did NOT fail", pod.Namespace, pod.Name, serialconfig.Config.SchedulerName)
 			}
 		})
+
+		It("[test_id:47619][tier3][unsched][default-scheduler] a deployment with a guaranteed pod resources available on one node but not on a single numa; scheduled by default scheduler", func() {
+
+			By("Scheduling the testing deployment")
+			deploymentName := "test-dp-with-default-sched"
+			var replicas int32 = 1
+
+			podLabels := map[string]string{
+				"test": "test-deployment-with-default-sched",
+			}
+			nodeSelector := map[string]string{}
+			deployment := objects.NewTestDeployment(replicas, podLabels, nodeSelector, fxt.Namespace.Name, deploymentName, objects.PauseImage, []string{objects.PauseCommand}, []string{})
+			// deployment is scheduled with the default scheduler
+			deployment.Spec.Template.Spec.SchedulerName = corev1.DefaultSchedulerName
+			deployment.Spec.Template.Spec.Containers[0].Resources.Limits = requiredRes
+
+			err := fxt.Client.Create(context.TODO(), deployment)
+			Expect(err).NotTo(HaveOccurred(), "unable to create deployment %q", deployment.Name)
+
+			By(fmt.Sprintf("checking deployment pods have been handled by the default scheduler %q but failed to be scheduled", corev1.DefaultSchedulerName))
+			pods, err := schedutils.ListPodsByDeployment(fxt.Client, *deployment)
+			Expect(err).NotTo(HaveOccurred(), "Unable to get pods from Deployment %q:  %v", deployment.Name, err)
+
+			for _, pod := range pods {
+				isFailed, err := nrosched.CheckPODSchedulingFailedForAlignment(fxt.K8sClient, pod.Namespace, pod.Name, corev1.DefaultSchedulerName, tmPolicy)
```

Comment:
here we expect a rejection from the kubelet - not from the scheduler, right?
If so we will need to add (and use here) a new helper, something like this
```
diff --git a/test/utils/nrosched/nrosched.go b/test/utils/nrosched/nrosched.go
index 9a358aa7..11f74465 100644
--- a/test/utils/nrosched/nrosched.go
+++ b/test/utils/nrosched/nrosched.go
@@ -39,16 +39,25 @@ import (
 const (
 	NROSchedObjectName = "numaresourcesscheduler"
 
+	// scheduler
 	ReasonScheduled        = "Scheduled"
 	ReasonFailedScheduling = "FailedScheduling"
+	// kubelet
+	ReasonTopologyAffinityError = "TopologyAffinityError"
 
+	// scheduler
 	ErrorCannotAlignPod       = "cannot align pod"
 	ErrorCannotAlignContainer = "cannot align container"
+	// kubeket
+	ErrorTopologyAffinityError = "Resources cannot be allocated with Topology locality"
+
+	// component name
+	kubeletName = "kubelet"
 )
 
 type eventChecker func(ev corev1.Event) bool
 
-func checkPODEvents(k8sCli *kubernetes.Clientset, podNamespace, podName, schedulerName string, evCheck eventChecker) (bool, error) {
+func checkPODEvents(k8sCli *kubernetes.Clientset, podNamespace, podName string, evCheck eventChecker) (bool, error) {
 	By(fmt.Sprintf("checking events for pod %s/%s", podNamespace, podName))
 	opts := metav1.ListOptions{
 		FieldSelector: fmt.Sprintf("involvedObject.name=%s", podName),
@@ -64,7 +73,7 @@ func checkPODEvents(k8sCli *kubernetes.Clientset, podNamespace, podName, schedul
 	}
 
 	for _, item := range events.Items {
-		klog.Infof("checking event: [%s: %s]", item.ReportingController, item.Reason)
+		klog.Infof("checking event: %s/%s [%s: %s - %s]", podNamespace, podName, item.ReportingController, item.Reason, item.Message)
 		if evCheck(item) {
 			klog.Infof("-> found relevant scheduling event for pod %s/%s: %v", podNamespace, podName, item)
 			return true, nil
@@ -78,7 +87,27 @@ func CheckPODSchedulingFailed(k8sCli *kubernetes.Clientset, podNamespace, podNam
 	isFailedScheduling := func(item corev1.Event) bool {
 		return item.Reason == ReasonFailedScheduling && item.ReportingController == schedulerName
 	}
-	return checkPODEvents(k8sCli, podNamespace, podName, schedulerName, isFailedScheduling)
+	return checkPODEvents(k8sCli, podNamespace, podName, isFailedScheduling)
+}
+
+func CheckPODKubeletRejectWithTopologyAffinityError(k8sCli *kubernetes.Clientset, podNamespace, podName string) (bool, error) {
+	isKubeletRejectForTopologyAffinityError := func(item corev1.Event) bool {
+		if item.Reason != ReasonTopologyAffinityError {
+			klog.Warningf("pod %s/%s reason %q expected %q", podNamespace, podName, item.Reason, ReasonTopologyAffinityError)
+			return false
+		}
+		// kubernetes is quircky and the component naming is a bit of hard to grok
+		if item.Source.Component != kubeletName {
+			klog.Warningf("pod %s/%s controller %q expected %q", podNamespace, podName, item.Source.Component, kubeletName)
+			return false
+		}
+		if !strings.Contains(item.Message, ErrorTopologyAffinityError) {
+			klog.Warningf("pod %s/%s message %q expected %q", podNamespace, podName, item.Message, ErrorTopologyAffinityError)
+			return false
+		}
+		return true
+	}
+	return checkPODEvents(k8sCli, podNamespace, podName, isKubeletRejectForTopologyAffinityError)
 }
 
 func CheckPODSchedulingFailedForAlignment(k8sCli *kubernetes.Clientset, podNamespace, podName, schedulerName, policy string) (bool, error) {
@@ -90,16 +119,29 @@ func CheckPODSchedulingFailedForAlignment(k8sCli *kubernetes.Clientset, podNames
 	}
 
 	isFailedSchedulingForAlignment := func(item corev1.Event) bool {
-		return item.Reason == ReasonFailedScheduling && item.ReportingController == schedulerName && strings.Contains(item.Message, alignmentErr)
+		if item.Reason != ReasonFailedScheduling {
+			klog.Warningf("pod %s/%s reason %q expected %q", podNamespace, podName, item.Reason, ReasonFailedScheduling)
+			return false
+		}
+		if item.ReportingController != schedulerName {
+			klog.Warningf("pod %s/%s controller %q expected %q", podNamespace, podName, item.ReportingController, schedulerName)
+			return false
+		}
+		// workaround kubelet race/confusing behaviour
+		if !strings.Contains(item.Message, alignmentErr) {
+			klog.Warningf("pod %s/%s message %q expected %q", podNamespace, podName, item.Message, alignmentErr)
+			return false
+		}
+		return true
 	}
-	return checkPODEvents(k8sCli, podNamespace, podName, schedulerName, isFailedSchedulingForAlignment)
+	return checkPODEvents(k8sCli, podNamespace, podName, isFailedSchedulingForAlignment)
 }
 
 func CheckPODWasScheduledWith(k8sCli *kubernetes.Clientset, podNamespace, podName, schedulerName string) (bool, error) {
 	isScheduledWith := func(item corev1.Event) bool {
 		return item.Reason == ReasonScheduled && item.ReportingController == schedulerName
 	}
-	return checkPODEvents(k8sCli, podNamespace, podName, schedulerName, isScheduledWith)
+	return checkPODEvents(k8sCli, podNamespace, podName, isScheduledWith)
 }
 
 func CheckNROSchedulerAvailable(cli client.Client, NUMAResourcesSchedObjName string) *nropv1alpha1.NUMAResourcesScheduler {
```

Reply:
Yeah, that's right. Will fix. Thanks for helping out and providing the code snippet.